### PR TITLE
test(api): add edge-case tests for empty contribution years

### LINKED ISSUE #237: test(api): add edge-case tests for empty contribution years
## 🧩 Background
Users with sparse or no recent activity can hit edge paths that are easy to miss without focused tests.

## 🎯 Objective
Strengthen test coverage for empty/missing contribution year data.

## 📁 Files to check
- `lib/calculate.test.ts`
- `app/api/streak/route.test.ts`
- `lib/github.test.ts`

## 🛠️ Implementation steps
- Add fixtures for empty years and partial-year data.
- Assert stable fallback metrics and non-crashing output.
- Verify API response behavior stays deterministic.

## ✅ Definition of done
- [ ] New tests cover empty and sparse year datasets.
- [ ] Existing green paths continue passing.
- [ ] No regressions in streak calculation behavior.

### PR DESCRIPTION
**Pull request by SiddharthRiot**
## Description
Adds regression tests for empty/sparse GitHub contribution-calendar edge cases to prevent crashes and ensure stable outputs for users with zero or minimal activity data. **Closes #237.**

## Pillar
Testing

## Changes
Added edge-case tests in `lib/calculate.test.ts` and `lib/github.test.ts` 
for empty and sparse contribution year data.

### `lib/calculate.test.ts`
- Returns stable output when all weeks have zero-contribution days
- Deterministic: same empty calendar always returns identical output
- Handles partial year (one week of data) without crashing

### `lib/github.test.ts`
- Handles calendar with all days having zero contributions
- Deterministic: two calls with empty-year response return identical data

## Checklist
- [x] npm run lint passes
- [x] npm run format passes  
- [x] npm run test passes (122 tests)

### CODE REVIEW

**Comment by SiddharthRiot** [comment]
hey @JhaSourav07! PR #251 is ready for review CI is passing and description has been updated with all required sections.

could you also add the relevant **labels**

and if anything needs to be fixed, do let me know and I'll get it done.

File: lib/calculate.test.ts [hunk 1]
```diff
@@ -258,3 +258,31 @@ describe('calculateStreak — timezone awareness', () => {
...
+
+  it('is deterministic: same empty calendar always returns identical output', () => {
+    const calendar = buildCalendar([]);
+    const r1 = calculateStreak(calendar);
+    const r2 = calculateStreak(calendar);
+    expect(r1).toEqual(r2);
+  });
+
+  it('handles partial year — only one week of data — without crashing', () => {
+    const calendar = buildCalendar([0, 1, 0, 0, 1, 0, 0]);
```
[the remark this answers is not in the document]
Comment (JhaSourav07) [reply lib/calculate.test.ts]: address this please

[the remark this answers is not in the document]
Comment (SiddharthRiot) [reply lib/calculate.test.ts]: > address this please

fixed the determinism test by passing a fixed `now` date (`2024-01-15T12:00:00Z`) to both `calculateStreak` calls, ensuring the test is not sensitive to UTC day boundaries.

**Comment by JhaSourav07** [comment]
can you keep the PR to focused on one commit.

**Comment by SiddharthRiot** [comment]
sure

**Review by Aamod007 — APPROVED** [review]
